f/connectors/alerts/alerts_gcs.py

```diff
@@ -337,34 +334,22 @@ def prepare_alerts_metadata(alerts_metadata, territory_id):
         A dictionary containing alert statistics: total alerts, month/year,
         and description of alerts.
     """
-    # Convert CSV bytes to DataFrame
+    # Convert CSV bytes to DataFrame and filter based on territory_id
     df = pd.read_csv(StringIO(alerts_metadata))
-
-    # Filter DataFrame based on territory_id
-    filtered_df = df.loc[df["territory_id"] == territory_id]
-
-    # Group the DataFrame by month and year, and get the last row for each group
-    filtered_df = (
-        filtered_df.loc[df["territory_id"] == territory_id]
-        .groupby(["month", "year"], as_index=False)
-        .last()
-    )
+    filtered_df = df.loc[df["territory_id"] == territory_id].copy()
 
     # Hash each row into a unique UUID; this will be used as the primary key for the metadata table
     # The hash is based on the most important columns for the metadata table, so that changes in other columns do not affect the hash
-    filtered_df["metadata_uuid"] = pd.util.hash_pandas_object(
-        filtered_df[
-            [
-                "territory_id",
-                "month",
-                "year",
```

Comment:
Adding the new column "description_alerts" to the object being hashed will change the UUIDs you generate. Will this ever be run on rows that already have a UUID in the DB?  (if not, then you are fine)

Reply:
Yes. We will have to re-populate the alerts metadata tables. Which is not a problem, I will do it after syncing the updated script.